tests/test_contracts.py

```
@@ -170,22 +185,245 @@ def test_decode_logs(self):
+            os.environ.update(saved_env)
+            if saved is None:
+                sys.modules.pop("ape", None)
+            else:
+                sys.modules["ape"] = saved
+
+    @pytest.mark.usefixtures("mock_load_abi")
+    def test_init_no_rpc_ape_import_error(self):
+        """No RPC_URL and ape import fails: plain ValueError (lines 166–173)."""
+        real_import = builtins.__import__
+
+        def _deny_ape(name, globals=None, locals=None, fromlist=(), level=0):
```

Ruff (0.15.9)

[warning] Missing return type annotation for private function `_deny_ape`

(ANN202)